conans/client/conan_api.py

```diff
@@ -456,6 +458,35 @@ def download(self, reference, remote_name=None, package=None, recipe=False):
         else:
             raise ConanException("Provide a valid full reference without wildcards.")
 
+    @api_method
+    def workspace_install(self, path, settings=None, options=None, env=None,
+                          remote_name=None, build=None, profile_name=None,
+                          update=False, cwd=None):
+        cwd = cwd or get_cwd()
+        if os.path.isabs(path):
+            abs_path = path
+        else:
+            abs_path = os.path.normpath(os.path.join(cwd, path))
```

Comment:
Why? You need to check if it is conditional? I don't understand.

Reply:
Yes, the `abs_path = os.path.normpath(os.path.join(cwd, path))` will work for already absolute paths.

Reply:
this ``os.path.normpath(os.path.join(cwd, path))`` is the same as using directly ``os.path.abspath(path)``. Wouldn't that work even if ``path`` is already absolute?

Reply:
Not the same, as long as ``cwd`` is an argument to the api.

Reply:
Since you are creating a new api method, is ``cwd`` parameter really needed?

Reply:
Yes, as all the other api methods, that already have ``cwd`` argument. 
We need to keep consistency. We could discuss over the conan_api as a whole somewhere else, but not here.

Reply:
There are some api methods that have a path parameter that do not have the ``cwd`` one. I don't see the need of having such a parameter in the api and here is not necessary either.

We could move the discussion to an engineering issue but I am still reluctant to include a new api method with a "useless" parameter...

Reply:
There are some API methods that the cwd really means something (for example, where to output files), and other that is not used at all, and only some path is needed. 

It is not useless, it means that the ``conanworkspace.cmake`` will be generated in that ``cwd`` if you use the python API, otherwise you need to do a chdir before the call (and undo it later, which is ugly and error prone). The ``cwd`` stays, until we decide to remove it from the whole api.